### PR TITLE
Improve passive templates error handling

### PR DESCRIPTION
**Pull request by xm1k3**
## Proposed changes

A change was made to the error message, it is now more explicit and is seen with the `--verbose` flag
> `template can't be used for offline matching`

Also performed checks on input validation to ensure that the user has entered either an URL or a file containing the domains. The error returned is `missing required input (http response) to run passive templates.`

## Checklist

- [x] Pull request is created against the [dev](https://github.com/projectdiscovery/nuclei/tree/dev) branch
- [ ] All checks passed (lint, unit/integration/regression tests etc.) with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by Mzack9999 — CHANGES_REQUESTED** [review]
Todo:
- [x] When passive flag is provided we ignore all templates which are not compatible, without posting misleading errors
- [x] If no input files are provided we error out as it happens now
- [ ] We just run compatible offline templates if any on the provided files

File: v2/pkg/catalog/loader/loader.go [hunk 1]
```diff
@@ -257,10 +257,10 @@ func areWorkflowTemplatesValid(store *Store, workflows []*workflows.WorkflowTemp
 }
 
 func isParsingError(message string, template string, err error) bool {
-	if err == templates.ErrCreateTemplateExecutor {
+	if err == filter.ErrExcluded {
```
Comment (Mzack9999) [inline v2/pkg/catalog/loader/loader.go:260]: `errors.Is(..)`

File: v2/pkg/catalog/loader/loader.go [hunk 2]
```diff
@@ -257,10 +257,10 @@ func areWorkflowTemplatesValid(store *Store, workflows []*workflows.WorkflowTemp
...
 		return false
 	}
-	if err == filter.ErrExcluded {
+	if err == templates.ErrCreateTemplateExecutor {
```
Comment (Mzack9999) [inline v2/pkg/catalog/loader/loader.go:263]: `errors.Is(..)`

File: v2/internal/runner/runner.go [hunk 3]
```diff
@@ -643,7 +648,9 @@ func (r *Runner) displayExecutionInfo(store *loader.Store) {
 	// Display stats for any loaded templates' syntax warnings or errors
 	stats.Display(parsers.SyntaxWarningStats)
 	stats.Display(parsers.SyntaxErrorStats)
-	stats.Display(parsers.RuntimeWarningsStats)
+	if r.options.Debug {
```
Comment (Mzack9999) [inline v2/internal/runner/runner.go:651]: Why are we hiding total runtime errors?

**Review by Mzack9999 — CHANGES_REQUESTED** [review]

**Review by Mzack9999 — APPROVED** [review]

**Comment by Mzack9999** [comment]
```console
$ go run . -passive

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   v2.8.6

                projectdiscovery.io

[INF] Using Nuclei Engine 2.8.6 (latest)
[INF] Using Nuclei Templates 9.3.4 (latest)
[INF] Templates added in last update: 15
[INF] Templates loaded for scan: 674
[INF] No results found. Better luck next time!
```

**Review by ehsandeep — APPROVED** [review]

**Pull request merged**